### PR TITLE
Code review request.

### PR DESCRIPTION
**Pull request by atenaiis**
Hello team,

This assignment consists of using the Devise gem to have a first approach at authentication in rails. You will build an application that allows users to create posts and the authors of the posts are displayed only if as a User you are logged in.

Please follow Readme file instructions.

Cheers!!!

### CODE REVIEW

File: app/views/posts/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    <div id="error_explanation">
+      <h2><%= pluralize(post.errors.count, "error") %> prohibited this post from being saved:</h2>
+
+      <ul>
+        <% post.errors.full_messages.each do |message| %>
+          <li><%= message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
```
Comment (oloomoses) [inline app/views/posts/_form.html.erb:12]: - [ ] KIndly Move this logic into a helper

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -2,15 +2,45 @@
...
+        <span class="glyphicon glyphicon-user text-white"> <%= link_to "Sign up", new_user_registration_path, :class => 'navbar-link'  %> |
+        <span class="glyphicon glyphicon-log-in text-white"> <%= link_to "Login", new_user_session_path, :class => 'navbar-link'  %>
+        <% end %>
+      </p>
+    </nav>
+    <div class="container-fluid">
+    <% if notice %>
+      <p class="alert alert-success"><%= notice %></p>
+    <% end %>
+    <% if alert %>
```
Comment (oloomoses) [inline app/views/layouts/application.html.erb:32]: - [ ] Kindly move this logic into a helper.

**Review by oloomoses — CHANGES_REQUESTED** [review]
# Changes Required :recycle: 
Hello @atenaiis, you have done a great job on this project.

### Highlight
- Most of the requirements are well implemented :muscle: 
- You have done very well on the styling and general look of the pages :100: 
- Great Job on the linters configuration, readme file and pull requests :clap: 
### To Improve

- [x] It would be nice to restrict your routes to `resources :posts, only: [:new, :create, :index]` as per the [requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png)
- [x] use **PostsController** **new** action to initialize a new post i.e `@post = Post.new` and not creating one
- [x] You have logic in your views. It would be once to move the into helpers as highlighted in the comment. Remember that view should strictly be used to render content from the corresponding controller and not to generate one.

Kindly implement the above changes before submitting another review.

**Feel free to reach me on slack or leave a comment while mentioning me and I will be happy to reply**

_Slack: Oloo Moses_

**Comment by atenaiis** [comment]
Thanks for the review.

I worked on changes requested.

cheers.

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -2,15 +2,45 @@
...
+      <p class="alert alert-danger"><%= alert %></p>
+    <% end %>
```
Comment (enaburekhan) [inline app/views/layouts/application.html.erb]: Move this logic to a helper. Views are only meant to present information passed to it by the corresponding controller. And not for generating it.

File: app/views/posts/_form.html.erb (hunk 1 above)
Comment (enaburekhan) [inline app/views/posts/_form.html.erb]: Also, move this logic to a helper.

**Review by enaburekhan — CHANGES_REQUESTED** [review]
# Status: Changes required:full_moon:
Hi Atenais!
**Great job so far:clap::clap:. Thank you for implementing some of the changes requested by the last reviewer. The codebase is clean and the app is working as expected.
However, we still need to make a few changes to get the project approved:muscle:.**

**To improve**
As highlighted by the last reviewer:

- [x] the application routes should use `resources :posts, only: [:new, :create, :index]` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication)). What the project currently has is just `resources :posts `. Kindly fix this.

- [x] Move the below highlighted logic in view to a helper.  There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it. Check this [article](http://codefol.io/posts/Where-Do-I-Put-My-Code/) and this [article](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers) if you need help on how to use a helper in rails. See [screen-shot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from pathwright.

Make the above-required changes and when you are ready, request another review.
Do have questions/comments? kindly drop them :point_down: and ping me on slack @ Eric Enaburekhan to respond.
Happy hacking:blush:
Eric Enaburekhan.

**Comment by atenaiis** [comment]
Thanks for that quick review.
I made the changes requested.
thank you.

Cheers!!!

**Review by cyonii — APPROVED** [review]
# Status: Approved 🎉✅

Hello team, you have shown a great effort so far. Great job implementing the requested changes.

## Nice to have
Please note that these are my suggestions, which means you do not have to implement them, however, I recommend them.
- [x] [OPTIONAL] Enclose pieces of code in your README in backticks `` ` ``, that way they stand out, e.g `bundle install`
- [x] [OPTIONAL] Since you have excluded only the `#update` method in your routes, I think it would be cleaner if you do this
```ruby
# Exclude only the one you don't want

resources :posts, except: [:update]
```
- [x] [OPTIONAL] You have `#update` action in your `PostsController` but you have removed its route from the configuration. Please remove actions and routes you are not using 
- [x] [OPTIONAL] It would be nice to make the logo clickable and takes the user to the root URL
- [x] Your application crashes when I want to log out, please fix this.
- [x] Your application crashes when I tried to edit posts.
- [x] The delete post functionality doesn't seem to be working


Happy coding, and keep your heads up 🤓💻

> #### <br>You can ping me on Slack @[CY Kalu](https://microverse-students.slack.com/team/U01494U96FR) to draw my attention in case I do not respond quickly on here. I'm always available and will be glad to help.<br><br>

**Pull request merged**